File: app/api/schemas/review.py
from datetime import datetime

from pydantic import BaseModel, Field, field_validator

from app.application.review_service import ReviewSectionData, ReviewSource


class UpdateSectionRequest(BaseModel):
    content: str = Field(min_length=1, max_length=10000)

    @field_validator("content")
    @classmethod
    def reject_blank_content(cls, value: str) -> str:
        if not value.strip():
            raise ValueError("板块内容不能为空")
        return value


class ReviewSectionResponse(BaseModel):
    section_key: str
    name: str
    revision: int
    content: str
    confirmed: bool
    editor: str
    created_at: datetime | None
    quality_status: str

    @classmethod
    def from_data(cls, data: ReviewSectionData) -> "ReviewSectionResponse":
        return cls(
            section_key=data.section.id,
            name=data.section.name,
            revision=data.review.revision,
            content=data.review.content,
            confirmed=data.review.confirmed,
            editor=data.review.editor,
            created_at=data.review.created_at,
            quality_status=data.generated.status,
        )


class ReviewSourceResponse(BaseModel):
    source_id: str
    fact_id: str
    person_id: str
    person: str
    line_start: int
    line_end: int
    excerpt: str

    @classmethod
    def from_source(cls, source: ReviewSource) -> "ReviewSourceResponse":
        return cls(**source.__dict__)

File: frontend/src/api/review.ts
import { api } from "./client";

export type ReviewSection = {
  section_key: string;
  name: string;
  revision: number;
  content: string;
  confirmed: boolean;
  editor: string;
  created_at: string | null;
  quality_status: string;
};

export type ReviewSource = {
  source_id: string;
  fact_id: string;
  person_id: string;
  person: string;
  line_start: number;
  line_end: number;
  excerpt: string;
};

export const reviewApi = {
  list: (taskId: string) => api<ReviewSection[]>(`/api/tasks/${taskId}/sections`),
  update: (taskId: string, sectionKey: string, content: string) => api<ReviewSection>(
    `/api/tasks/${taskId}/sections/${sectionKey}`,
    { method: "PUT", headers: { "Content-Type": "application/json" }, body: JSON.stringify({ content }) },
  ),
  confirm: (taskId: string, sectionKey: string) => api<ReviewSection>(
    `/api/tasks/${taskId}/sections/${sectionKey}/confirm`, { method: "POST" },
  ),
  versions: (taskId: string, sectionKey: string) => api<ReviewSection[]>(
    `/api/tasks/${taskId}/sections/${sectionKey}/versions`,
  ),
  restore: (taskId: string, sectionKey: string, revision: number) => api<ReviewSection>(
    `/api/tasks/${taskId}/sections/${sectionKey}/restore/${revision}`, { method: "POST" },
  ),
  regenerate: (taskId: string, sectionKey: string, instruction: string) => api<ReviewSection>(
    `/api/tasks/${taskId}/sections/${sectionKey}/regenerate`,
    { method: "POST", headers: { "Content-Type": "application/json" }, body: JSON.stringify({ instruction }) },
  ),
  sources: (taskId: string, sectionKey: string) => api<ReviewSource[]>(
    `/api/tasks/${taskId}/sections/${sectionKey}/sources`,
  ),
};
